packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -244,11 +261,23 @@ export class SerializedStateManager {
 				}
 				assert(this.snapshot !== undefined, 0x8e5 /* no base data */);
 				const pendingRuntimeState = await runtime.getPendingLocalState(props);
+				// This conversion is required because ArrayBufferLike doesn't survive JSON.stringify
+				const loadedGroupIdSnapshots = {};
+				if (this.storageAdapter.loadedGroupIdSnapshots) {
+					for (const [groupId, snapshot] of Object.entries(
+						this.storageAdapter.loadedGroupIdSnapshots,
+					)) {
+						loadedGroupIdSnapshots[groupId] = convertSnapshotToSnapshotInfo(snapshot);
+					}
+				}
 				const pendingState: IPendingContainerState = {
 					attached: true,
 					pendingRuntimeState,
 					baseSnapshot: this.snapshot.baseSnapshot,
 					snapshotBlobs: this.snapshot.snapshotBlobs,
+					loadedGroupIdSnapshots: this.storageAdapter.loadedGroupIdSnapshots
```

Comment:
Maybe we should check that the length of the "loadedGroupIdSnapshots" is more than 0 and not just undefined. That will make this check more robust.